### PR TITLE
Update tree-sitter-python + Add missing tree-sitter scopes

### PR DESCRIPTION
**Pull request by aminya**
### Description of the Change

This updates tree-sitter python dependency.

### Benefits

Bug fixes: https://github.com/tree-sitter/tree-sitter-python/compare/v0.15.1...v0.17.0

### Applicable Issues
Solves some of the open issues regarding syntax highlighting.

Fixes https://github.com/atom/language-python/issues/324
Fixes https://github.com/atom/language-python/issues/134
Fixes https://github.com/atom/language-python/issues/273
Fixes https://github.com/atom/language-python/issues/312

Allows removing the Kite's workaround for tree-sitter: 
https://github.com/kiteco/atom-plugin/pull/704

### CODE REVIEW

**Comment by aminya** [comment]
@sadick254 @darangi Please take a look.

File: grammars/tree-sitter-python.cson [hunk 1]
```diff
@@ -73,18 +73,21 @@ scopes:
   'interpolation > "}"': 'punctuation.section.embedded'
 
   'class_definition > identifier': 'entity.name.type.class'
-  'function_definition > identifier': 'entity.name.function'
+  'function_definition > identifier': 'entity.name.function.def'
```
Comment (jeff-hykin) [inline grammars/tree-sitter-python.cson:76]: this should probably be `entity.name.function.definition` to be more consistent with other languages

Comment (aminya) [reply grammars/tree-sitter-python.cson:76]:
```suggestion
  'function_definition > identifier': 'entity.name.function.definition'
```

Comment (aminya) [reply grammars/tree-sitter-python.cson]: Thanks! fixed.

**Comment by chbk** [comment]
As a heads-up, #313 also fixes the missing scopes mentioned here. That said, it's part of a larger ecosystem of PRs that might take time to (hopefully) get merged, so I'm glad to see this fixed now.

**Comment by aminya** [comment]
> As a heads-up, #313 also fixes the missing scopes mentioned here. That said, it's part of a larger ecosystem of PRs that might take time to (hopefully) get merged, so I'm glad to see this fixed now.

Thanks for the PR. I only need a few of these commits in my PR so I can close many things here and there. If you can rebase after me, that would be great.

**Comment by ThatXliner** [comment]
What’s the status on this?

**Comment by aminya** [comment]
This is ready to go. We are awaiting one of the maintainers to merge this.
@sadick254 @darangi

**Comment by ThatXliner** [comment]
@jeff-hykin (the reviewer)

**Comment by jeff-hykin** [comment]
@ThatXliner Not sure why I'm reviewer, but I added a review 😁

**Comment by darangi** [comment]
Thanks @aminya , We will look into this

**Comment by darangi** [comment]
@aminya could you resolve this conflict so I merge this in? Thanks for your contributions 🙇🏾

**Comment by aminya** [comment]
Thank you! The conflict was only the deleted travis file.

**Comment by jeff-hykin** [comment]
(I just realized I never clicked "submit" when I said "but I added a review" 😬)

**Comment by aminya** [comment]
Thanks for the review. Do you want to make a PR to fix these?

**Comment by jeff-hykin** [comment]
The first comment should probably be a PR, although that edgecase bug I'm describing probably happens other places too. Its a really common thing.

**Review by MdAbulHossain1 — COMMENTED** [review]
@aminya

**Pull request merged**